src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2071,3 +2077,148 @@
    descPath <- file.path(packagePath, "DESCRIPTION")
    read.dcf(descPath, all = TRUE)
 })
+
+.rs.addFunction("readSourceDocument", function(id)
+{
+   contents <- .Call("rs_readSourceDocument", as.character(id), PACKAGE = "(embedding)")
+   Encoding(contents) <- "UTF-8"
+   contents
+})
+
+.rs.addFunction("discoverPackageDependencies", function(id, type)
+{
+   # check to see if we have available packages -- if none, bail
+   available <- .rs.availablePackages()
+   if (is.null(available$value))
+      return(character())
+   
+   # attempt to read the file
+   contents <- .rs.tryCatch(.rs.readSourceDocument(id))
```

Comment:
It looks like this is expecting `.rs.readSourceDocument` to raise an error but as far as I can tell it can't do this at present (instead it returns null if it can't read the doc).